dbms/src/Storages/DeltaMerge/tests/gtest_dm_vector_index.cpp

```diff
@@ -999,34 +1348,50 @@ try
 
     // Pack Filter using RowKeyRange
     {
-        const auto ann_query_info = createANNQueryInfo(vec_cd.id, tipb::VectorDistanceMetric::L2, 1, {8.0});
+        auto ann_query_info = std::make_shared<tipb::ANNQueryInfo>();
+        ann_query_info->set_column_id(vec_cd.id);
+        ann_query_info->set_distance_metric(tipb::VectorDistanceMetric::L2);
+        ann_query_info->set_top_k(1);
+        ann_query_info->set_ref_vec_f32(encodeVectorFloat32({8.0}));
 
         // This row key range will cause pack#0 and pack#1 reserved, and pack#2 filtered out.
         auto row_key_ranges = RowKeyRanges{RowKeyRange::fromHandleRange(HandleRange(0, 5))};
 
         auto bitmap_filter = std::make_shared<BitmapFilter>(9, false);
         bitmap_filter->set(0, 6); // 0~6 rows are valid, 6~9 rows are invalid due to pack filter.
 
+        auto vec_idx_ctx = VectorIndexStreamCtx::createForStableOnlyTests(
+            ann_query_info,
+            std::make_shared<ColumnDefines>(read_cols));
         DMFileBlockInputStreamBuilder builder(dbContext());
-        auto stream = builder.setAnnQureyInfo(ann_query_info)
-                          .setBitmapFilter(BitmapFilterView(bitmap_filter, 0, 9))
+        auto stream = builder.setVecIndexQuery(vec_idx_ctx)
                           .build(dm_file, read_cols, row_key_ranges, std::make_shared<ScanContext>());
         ASSERT_INPUTSTREAM_COLS_UR(
-            stream,
+            VectorIndexTestUtils::wrapVectorStream( //
+                vec_idx_ctx,
+                stream,
+                bitmap_filter),
             createColumnNames(),
             createColumnData({
                 createColumn<Int64>({5}),
                 createVecFloat32Column<Array>({{5.0}}),
             }));
 
         // TopK=4
-        const auto ann_query_info2 = createANNQueryInfo(vec_cd.id, tipb::VectorDistanceMetric::L2, 4, {8.0});
+        ann_query_info = std::make_shared<tipb::ANNQueryInfo>();
```

Comment:
I reverted this shortcut, because we are soon going to support MaxDistance, and WHERE push down, which will make this `createANNQueryInfo` very hard to understand. However, the TSAN issue is still resolved, by manually creating a new ANNQueryInfo. What do you think of this change? @Lloyd-Pottiger